src/shared/components/TextExpander.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import {observer} from "mobx-react";
+import {If, Else, Then} from 'react-if';
+import TextTruncate from 'react-text-truncate';
+import {observable} from "mobx";
+
+export interface ITextExpanderProps
+{
+    text?: string|JSX.Element|null;
+    line?: number;
+    expandIndicator?: string|JSX.Element;
+    collapseIndicator?: string|JSX.Element;
+    truncateText?: string;
+}
+
+@observer
+export default class TextExpander extends React.Component<ITextExpanderProps, {}> {
+    @observable protected isTextTruncated: boolean = true;
```

Comment:
I've been preferring mobx to setState, my understanding/impression was that it replaces it, and thus does fully mobx-ifies the react component lifecycle